### PR TITLE
Unified URL widget, non-blocking URL validation and Exporter support for remote output databases

### PR DESCRIPTION
**Pull request by soininen**
This PR contains three linked changes:

- It is now possible to specify a fixed database URL for output in Exporter. This enables writing to remote databases.
- Data Store, Data Connection, Importer and Exporter (which now needs the widget too) share the same URL widget. Previously, Data Store's properties widget was still using its own URL controls.
- Exporter doesn't really need the heavy database validation that came with how we converted URLs to the SQLAlchemy format. Therefore, checking that the database actually exists and is valid is now done as-needed and in a separate thread as the operation used to block Toolbox UI for long time when remote databases were used.

This PR depends on spine-tools/Spine-Database-API#230 and spine-tools/Spine-Toolbox#2028

Resolves spine-tools/Spine-Toolbox#1668

## Checklist before merging
- [x] Documentation (also in Toolbox repo) is up-to-date
- [x] Release notes in Toolbox repo have been updated
- [x] Unit tests have been added/updated accordingly
- [x] Code has been formatted by black
- [x] Unit tests pass

### CODE REVIEW

File: spine_items/database_validation.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        self._signals = _TaskSignals()
+        self._signals.validation_failed.connect(fail_slot)
+        if success_slot is not None:
+            self._signals.validation_succeeded.connect(success_slot)
+        self._signals.finished.connect(finish_slot)
+
+    def run(self):
+        """Checks if it is possible to establish an SQLAlchemy connection to the given URL."""
+        try:
+            self._signals.moveToThread(QThread.currentThread())
```
Comment (manuelma) [inline spine_items/database_validation.py:44]: I'm getting the below at loading a project that has Data Stores?
```
QObject::moveToThread: Current thread (0x5570c5237ce0) is not the object's thread (0x5570c4907db0).
Cannot move to target thread (0x5570c5237ce0)
```

Comment (soininen) [reply spine_items/database_validation.py:44]: Interesting. I don't see this, could it be something Linux specific? In any case, I'll have a look.

Comment (soininen) [reply spine_items/database_validation.py:44]: Haha, I should read the documentation more carefully:

> Warning: This function is not thread-safe; the current thread must be same as the current thread affinity. In other words, this function can only "push" an object from the current thread to another thread, it cannot "pull" an object from any arbitrary thread to the current thread. There is one exception to this rule however: objects with no thread affinity can be "pulled" to the current thread.

(from [Qt documentation](https://doc.qt.io/qt-6/qobject.html#moveToThread))

Sorry for the inconvenience! For some reason this worked perfectly fine on Windows. I'll try to get it fixed ASAP.

Comment (soininen) [reply spine_items/database_validation.py:44]: @manuelma Could you try if `1668_moveToThread_issue` branch fixes the Data Store issue for you? I cannot reproduce the Traceback here. Meanwhile, I'll try to come up with a unit test that catches the Traceback in GitHub actions.

**Pull request merged**